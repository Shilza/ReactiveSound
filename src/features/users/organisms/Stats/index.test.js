import React from 'react';
import {cleanup, render} from 'react-testing-library';
import {Stats} from "./index";
import {BrowserRouter as Router} from "react-router-dom";

describe('<Stats/> users feature organism', () => {
    const user = {
        id: 5
    };
    const loading = false;
    const location = {
        pathname: 'tracks'
    };
    const {container, getByTestId} = render(
        <Router>
            <Stats
                user={user}
                loading={loading}
                location={location}
            />
        </Router>
    );

    it('should have child', () => {
        expect(container.firstChild.childNodes.length).toBeGreaterThanOrEqual(1);
    });

    it('should have correct user tracks link', () => {
        expect(getByTestId('userTracksLink')).toHaveAttribute('href', `/users/${user?.id}/tracks`);
    });

    it('should have correct user\'s likes tracks link', () => {
        expect(getByTestId('userLikedTracksLink')).toHaveAttribute('href', `/users/${user?.id}/liked`);
    });

    it('should not have child', () => {
        cleanup();

        const loading = true;
        const {container} = render(
            <Router>
                <Stats
                    user={user}
                    loading={loading}
                    location={location}
                />
            </Router>
        );

        expect(container.firstChild.childNodes.length).toBe(0);
    });
});