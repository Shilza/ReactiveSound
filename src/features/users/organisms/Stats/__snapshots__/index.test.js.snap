// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats/> should renders correctly 1`] = `
<div
  class="container"
>
  <div
    class="unit"
  >
    <a
      class="label"
      data-testid="userTracksLink"
      href="/users/5/tracks"
    >
      Tracks
    </a>
    <span
      class="count"
    >
      0
    </span>
  </div>
  <div
    class="unit"
  >
    <a
      class="label"
      data-testid="userLikedTracksLink"
      href="/users/5/liked"
    >
      Likes
    </a>
    <span
      class="count"
    >
      0
    </span>
  </div>
  <div
    class="unit"
  >
    <div
      class="label"
    >
      Followers
    </div>
    <span
      class="count"
    >
      0
    </span>
  </div>
  <div
    class="unit"
  >
    <div
      class="label"
    >
      Followings
    </div>
    <span
      class="count"
    >
      0
    </span>
  </div>
</div>
`;
